frontend/src/modules/notification/actions.tsx

```diff
@@ -0,0 +1,54 @@
+import React from "react"
+import { AppDispatch, RootState } from "../../store"
+import { NOTIFICATION_REMOVE, NotificationAction as NotificationAction, NotificationID, NotificationProps, NotificationRemoveAction as NotificationRemoveAction, NOTIFICATION, NotificationType } from "./models"
+import { notification } from "antd"
+
+type AntdNotificationType = keyof Pick<typeof notification, 'success' | 'info' | 'warning' | 'error'>
+const convertToAntdNotificationType: { [key in NotificationType]: AntdNotificationType } = {
+    [NotificationType.SUCCESS]: 'success',
+    [NotificationType.INFO]: 'info',
+    [NotificationType.WARNING]: 'warning',
+    [NotificationType.ERROR]: 'error'
+}
+
+const hasNotification = (state: RootState, id: NotificationID) => {
+    return state.notifications.some((notification) => notification.id === id)
+}
+
+export const notify = (id: NotificationID, props: NotificationProps) => async (dispatch: AppDispatch, getState: () => RootState) => {
+    if (hasNotification(getState(), id)) {
```

Comment:
Should close existing notification and replace with the new instance at the head of the list.